### DISCUSSION TITLE
Unrecognized pair style 'deepmd'

### DISCUSSION THREAD

**Discussion by EricGaoOm** [deepmodeling/deepmd-kit#4211] [General]
I was running md in DPA2, an warning occured like this 
I was curious about such warning caused by incompatible version of lammps and deepmd-kit or other reasons
![Uploading 111.png…]()

**Comment by EricGaoOm** [comment]
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:275)
Last command: pair_style        deepmd dpa2_model.pth

**Comment by njzjz** [comment]
Could you list how you install the program, how you run the program, and the entire output of the program?

**Reply by 96XinHe** [reply]
I'm experiencing the same problem. I downloaded the offline packages and installed them via the sh command (both vesion 3.0.0b4 and 3.0.0rc0). Subsequent simulations using lammps gave the same error, whether using the se_a2 I had previously trained with TF in the old deepmd-kit or the newly trained DPA-2 model: 
```
LAMMPS (29 Aug 2024)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Open of file /data/run01/scw6dyq/software/DP_series/deepmd-kit_3.0.0rc0/lib/deepmd_lmp/dpplugin.so failed: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /data/run01/scw6dyq/software/DP_series/deepmd-kit_3.0.0rc0/bin/../lib/./python3.10/site-packages/tensorflow/../../.././libcares.so.2)
Loaded 0 plugins from /data/run01/scw6dyq/software/DP_series/deepmd-kit_3.0.0rc0/lib/deepmd_lmp
Reading data file ...
  triclinic box = (0 0 0) to (59.0364 51.127022 55.6378) with tilt (-29.5182 0 0)
WARNING: Triclinic box skew is large. LAMMPS will run inefficiently. (src/domain.cpp:221)
  1 by 1 by 1 MPI processor grid
  reading atom labelmap ...
  reading atoms ...
  15876 atoms
  read_data CPU = 0.049 seconds
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:275)
Last command: pair_style      deepmd freeze_model.pth
```
If I use an older version of deepmd-kit 2.2.11 that I installed using the same method, I can read the .pb model and run the simulation just fine.


I notice the erro information:
`version 'GLIBC_2.25' not found (required by /data/run01/scw6dyq/software/DP_series/deepmd-kit_3.0.0rc0/bin/../lib/./python3.10/site-packages/tensorflow/../../.././libcares.so.2)`
and i checked my version by
`ldd --version
`
```
ldd (GNU libc) 2.17
Copyright (C) 2012 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```
It looks like the gnu version is too low on the platform I'm using right? Is it possible to include the gnu version in the next update package?

**Reply by njzjz** [reply]
Related to https://github.com/conda-forge/c-ares-feedstock/pull/38#issuecomment-2284724240

**Reply by njzjz** [reply]
You can downgrade `c-ares` by yourself.

The next version of the offline packages will ship an old version of `c-ares`.

**Reply by 96XinHe** [reply]
Thanks, I’ve successfully downgraded  `c-ares` from version 1.34.3 to 1.32.3 using the following commands:
```
conda config --add channels conda-forge
conda config --set channel_priority strict
conda install c-ares=1.32.3
```
Now, the `lmp` command no longer reports an error.

**Reply by EricGaoOm** [reply]
Thanks, I will try this solution!
